grakn-engine/src/main/java/ai/grakn/engine/task/BackgroundTask.java

```diff
@@ -16,20 +16,31 @@
  * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
  */
 
-package ai.grakn.engine.tasks.mock;
-
-import ai.grakn.engine.TaskId;
+package ai.grakn.engine.task;
 
 /**
- * Mocked task that will throw exception
+ * <p>
+ *     Defines the API which must be implemented in order to be able to run the task in the background
+ * </p>
  *
- * @author alexandraorth, Felix Chapman
+ * @author Filipe Peliz Pinto Teixeira
  */
-public class FailingMockTask extends MockBackgroundTask {
+public interface BackgroundTask {
```

Comment:
I think  you know this, but we only have one `BackgroundTask` now. I guess this lets us prepare for the future.